Add a `.gte()` method for checking if a version is greater than or equal to another

LucianBuzzo opened the pull request:
Connects to #15 

change-type: minor

pimterry commented on src/index.ts:
```diff
@@ -150,3 +152,54 @@ export const major = (version: string | null): number | null => {
+		return true;
+	}
+
+	return semver.gte(versionA, versionB);
```
This has inconsistent behaviour with `compare`, which takes `dev` and `rev` suffixes into account. I think what should probably happen is that `gte` should be equivalent to `compare(a, b) >= 0`. Is there any reason those should give different results?

LucianBuzzo replied:
Yes, I missed this. These comparison methods (lt, gt, gte etc) should be using compare internally. I'll add some test cases for dev and rev as well 👍

LucianBuzzo commented:
@pimterry I've added comprehensive testing for the `.compare()` function and fixed the implementation as it was previously performing a right compare, meaning the results were the opposite of what you would expect. The `.gte()` function now wraps `.compare()`. I'll propagate these changes to the other PRs once this one is merged.

pimterry commented on src/index.ts:
```diff
@@ -189,3 +197,29 @@ export const prerelease = (version: string | null) => {
+ * @example
+ * resinSemver.gte('2.0.5', '1.16.0'); //--> true
+ *
+ * resinSemver.gte('Resin OS 2.0.5', 'Resin OS 2.0.2+rev2'); //--> false
```
I think this example is wrong - this will (correctly) return `true`
